From e926f67c2740357406fad26fee53487259b85c27 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Fri, 18 Mar 2016 06:06:04 -0700
Subject: [PATCH 1624/1767] Revert "UBUNTU: SAUCE: apparmor: Fix: ensure
 aa_get_newest will trip debugging if the replacedby is not setup"

BugLink: http://bugs.launchpad.net/bugs/1379535

This reverts commit 9b02a4b233d1b5b9ec3cb4353b894089a1bf9ea0.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 security/apparmor/include/label.h | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/security/apparmor/include/label.h b/security/apparmor/include/label.h
index 3530efe..071eabe 100644
--- a/security/apparmor/include/label.h
+++ b/security/apparmor/include/label.h
@@ -384,17 +384,8 @@ static inline struct aa_label *aa_get_newest_label(struct aa_label *l)
 	if (!l)
 		return NULL;
 
-	if (label_invalid(l)) {
-		struct aa_label *tmp;
-		AA_BUG(!l->replacedby);
-		AA_BUG(!l->replacedby->label);
-		/* BUG: only way this can happen is @l ref count and its
-		 * replacement count have gone to 0 and are on their way
-		 * to destruction. ie. we have a refcounting error
-		 */
-		AA_BUG(!(tmp = aa_get_label_rcu(&l->replacedby->label)));
-		return tmp;
-	}
+	if (label_invalid(l))
+		return aa_get_label_rcu(&l->replacedby->label);
 
 	return aa_get_label(l);
 }
-- 
1.9.3

